### PR TITLE
Expando verification for programs, shaders, renderbuffers, textures and ...

### PR DESCRIPTION
**Pull request by RafaelCintron**
...framebuffers

Ensure that any expandos set on programs, shaders, renderbuffers, textures and framebuffers
remain set across garbage collections, even when the only references are internal to WebGL.

### CODE REVIEW

File: sdk/tests/conformance/misc/expando-loss.html [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    var fs = wtu.setupSimpleColorFragmentShader(gl);
+    setTestExpandos(fs);
+
+    var program = wtu.setupProgram(gl, [vs, fs]);
+    setTestExpandos(program);
+    assertMsg(program === gl.getParameter(gl.CURRENT_PROGRAM), "getParameter(gl.CURRENT_PROGRAM) return instance set with useProgram");
+
+    var attachedShaders = gl.getAttachedShaders(program);
+    assertMsg(vs === attachedShaders[0], "getAttachedShaders[0] returns instance set with attachShader");
```
Comment (zhenyao) [inline sdk/tests/conformance/misc/expando-loss.html]: To my knowledge, this order (0 -> vertex shader, 1 ->fragment shader) is not guaranteed.

Comment (kenrussell) [inline sdk/tests/conformance/misc/expando-loss.html]: Agreed; this order doesn't seem to be specified in the ES 2.0 spec. Would be safer to write:

assertMsg(vs === attachedShaders[0] || vs === attachedShaders[1], ...);
assertMsg(fs === attachedShaders[0] || fs === attachedShaders[1], ...);

**Comment by kenrussell** [comment]
Nice test overall. Could you address the one comment above? Will gladly merge afterward. Thanks.

**Comment by RafaelCintron** [comment]
@kenrussell and @zhenyao , you're right about getAttachedShaders.  I've uploaded a new change that addresses this issue. Let me know if there's more you'd like me to improve.

**Comment by kenrussell** [comment]
Thanks for the update. Looks great.

**Pull request merged**